src/huggingface_hub/repository.py

```diff
@@ -112,6 +112,29 @@ def is_git_ignored(filename: Union[str, Path]) -> bool:
     return is_ignored
 
 
+def is_tracked_upstream(folder: Union[str, Path]) -> bool:
+    """
+    Check if the current checked-out branch is tracked upstream.
+    """
+    try:
+        p = subprocess.run(
+            ["git", "status", "-sb"],
+            stderr=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            encoding="utf-8",
+            cwd=folder,
+        )
+        status = p.stdout
+    except subprocess.CalledProcessError as exc:
+        raise OSError(exc.stderr)
+
+    if "HEAD" in status:
+        raise OSError("No branch checked out.")
+
+    is_tracked = ".." in status
```

Comment:
Could you expand a bit on this one? Is it because in the format of the status you get something like this?:

`## master...origin/master` 

I wonder if there's something a bit more error-proof, such as `git rev-parse --symbolic-full-name --abbrev-ref @{u}`, but this one is also harder to maintain/understand (this command gives a clear fatal error when there is no upstream configured).